fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java

```diff
@@ -74,18 +69,15 @@ public class HttpServerHandler extends ChannelInboundHandlerAdapter {
     private final LongCounterMetric handlingRequestsNum;
     private final Histogram requestHandleLatencyMs;
 
-    public HttpServerHandler(ActionController controller) {
+    public HttpServerHandler(ActionController controller,
+                             LongCounterMetric httpConnectionsNum,
+                             LongCounterMetric handlingRequestsNum,
+                             Histogram requestHandleLatencyMs) {
```

Comment:
Although this is doable to fix the duplicate metrics registration issue. It doesn't look elegant to pass three parameters who are not significant to the core function of the HttpServerHandler.

The interface can't be stable as more metrics to be added in future.

I am thinking if the HttpMetricRegistry can add an interface called `getOrRegisterXXX()` so each handler can call this interface to avoid registering the same metrics again and again.

Reply:
Yes, passing parameters directly is not extensible. Now I use `HttpServerHandlerMetrics#INSTANCE` to ensure only register metrics once. I don't use `getOrRegisterXXX` because the interface need to check each time, and each handler also need to tell the interface how to create a metric if not exist. I think using instance can make it more simple.